// react-marked-markdown/dist/components/MarkdownInput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getPrototypeOf = require("babel-runtime/core-js/object/get-prototype-of");

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require("babel-runtime/helpers/classCallCheck");

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require("babel-runtime/helpers/createClass");

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require("babel-runtime/helpers/possibleConstructorReturn");

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require("babel-runtime/helpers/inherits");

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var MarkdownInput = function (_React$Component) {
  (0, _inherits3.default)(MarkdownInput, _React$Component);

  function MarkdownInput() {
    (0, _classCallCheck3.default)(this, MarkdownInput);
    return (0, _possibleConstructorReturn3.default)(this, (0, _getPrototypeOf2.default)(MarkdownInput).apply(this, arguments));
  }

  (0, _createClass3.default)(MarkdownInput, [{
    key: "render",
    value: function render() {
      var _props = this.props;
      var onChange = _props.onChange;
      var value = _props.value;
      var placeholder = _props.placeholder;
      var className = _props.className;

      return _react2.default.createElement("textarea", {
        onChange: onChange,
        value: value,
        placeholder: placeholder,
        ref: "textareaRef",
        className: className
      });
    }
  }]);
  return MarkdownInput;
}(_react2.default.Component);

exports.default = MarkdownInput;


MarkdownInput.propTypes = {
  onChange: _react2.default.PropTypes.func,
  value: _react2.default.PropTypes.string,
  placeholder: _react2.default.PropTypes.string,
  className: _react2.default.PropTypes.string
};
